src/net_processing.cpp

```diff
@@ -3305,7 +3305,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         {
             LOCK(cs_main);
 
-            const CBlockIndex* pindex = m_chainman.m_blockman.LookupBlockIndex(req.blockhash);
+            const CBlockIndex* pindex{LookupBlockIndexForPeer(pfrom, req.blockhash)};
```

Comment:
How could one have used `BLOCKTXN` for fingerprinting considering that we'd not serve data for blocks with a height difference of <10 to our tip (`MAX_BLOCKTXN_DEPTH`) anyway?

Reply:
Good that you made me double check, because this can actually not be used for fingerprinting but for a different reason than the one you mentioned.

If the requested block is older than `MAX_BLOCKTXN_DEPTH` blocks, we treat it as a normal `getdata` request for that block see: https://github.com/bitcoin/bitcoin/blob/8c721fff3a007db613b22464b40fe7cd6b1ead80/src/net_processing.cpp#L3278-L3287

So *i thought* the fingerprint would be:
If the block exists in the nodes index, it will respond with `blocktxn` or `block` depending on the height.
If the block does not exist, it won't respond at all.

However, when we process block requests (`ProcessGetBlockData()`) we check for non main chain requests using `BlockRequestAllowed()`, so `GETBLOCKTXN` can not be abused for fingerprinting.
https://github.com/bitcoin/bitcoin/blob/8c721fff3a007db613b22464b40fe7cd6b1ead80/src/net_processing.cpp#L1933-L1936

I will remove the commit.

Reply:
@mzumsande and I discussed this yesterday and we came to the same conclusion - that this could be used to determine what stale blocks the peer has that are older than 10 blocks but less than 30 days old. That's not a fingerprint attack.

I think that we can probably just remove the whole logic below "If an older block is requested...":

https://github.com/bitcoin/bitcoin/blob/b4f686952a60bbadc7ed2250651d0d6af0959f4d/src/net_processing.cpp#L3278-L3289)

and exit early if the block is deeper than 10 deep.

The rationale of using outbound bandwidth as DoS mitigation doesn't seem very good to me, and if we return early because the block is older than 10 deep, then we don't have an expensive disk read (we just look up the block index in the in-memory BlockMap and then return immediately). I think that can be done in a follow-up PR.